src/services/database.ts

```
@@ -0,0 +1,718 @@
+          throw err;
+        }
+        // Handle network/fetch errors
+        if (err.name === 'TypeError' && err.message.includes('fetch')) {
+          throw new Error('Unable to connect to database. Please check your internet connection and Supabase configuration.');
+        }
+      }
+      // Re-throw unknown errors
+      throw new Error(`Failed to update chat: ${err instanceof Error ? err.message : 'Unknown error'}`);
+    }
+  }
+}
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)